### PR TITLE
[SYNCOPE-1665] show error on propagation failure, made some refactoring

### PR DESCRIPTION
**Pull request by andrea-patricelli**

### CODE REVIEW

File: client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/commons/RESTUtils.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+package org.apache.syncope.client.enduser.commons;
+
+import org.apache.syncope.client.enduser.rest.UserSelfRestClient;
+import org.apache.syncope.common.lib.request.UserCR;
+import org.apache.syncope.common.lib.request.UserUR;
+import org.apache.syncope.common.lib.to.ProvisioningResult;
+import org.apache.syncope.common.lib.to.UserTO;
+
+public final class RESTUtils {
```
Comment (ilgrosso) [inline client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/commons/RESTUtils.java:28]: This class seems to be some sort of handler for `ProvisioningResult`, please name it accordingly,

File: client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/commons/PageParametersUtils.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.syncope.client.enduser.SyncopeWebApplication;
+import org.apache.syncope.client.enduser.pages.Dashboard;
+import org.apache.syncope.client.ui.commons.Constants;
+import org.apache.syncope.common.lib.to.PropagationStatus;
+import org.apache.wicket.Component;
+import org.apache.wicket.request.mapper.parameter.PageParameters;
+
+public final class PageParametersUtils {
```
Comment (ilgrosso) [inline client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/commons/PageParametersUtils.java:30]: This seems to be related only to propagation, please merge it with class below.

File: client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/SyncopeEnduserApplication.properties [hunk 3]
```diff
@@ -78,6 +78,7 @@ before=Before
 after=After
 
 captcha_error=Captcha entered does not match
+blank_security_info=Security question and/or answer is/are blank
```
Comment (ilgrosso) [inline client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/SyncopeEnduserApplication.properties:81]: Is this key not needed in other languages?

File: client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult_ja.properties [hunk 4]
```diff
@@ -14,4 +14,6 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-login=Back to Home
+login=????
```
Comment (ilgrosso) [inline client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult_ja.properties:17]: Are those real question marks?

File: client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult_ru.properties [hunk 5]
```diff
@@ -14,4 +14,6 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-login=Back to Home
+login=????????? ?????
```
Comment (ilgrosso) [inline client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult_ru.properties:17]: Are those real question marks?

**Review by ilgrosso — CHANGES_REQUESTED** [review]
I appreciate the overall idea, but, besides the changes already requested, I think it would be useful to:

- [x] add two properties (not one) as of `reportPropagationErrors` and `reportPropagtionErrorDetails`: the former will just enable the propagation error reporting with a generic message as 'Something went wrong during propagation, contact your administrator', while the latter will allow to explore the actual error per Resource
- [x] definitely improve UI

File: client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/SyncopeEnduserApplication.properties (hunk 3 above)
Comment (andrea-patricelli) [reply client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/SyncopeEnduserApplication.properties:81]: yes, sure. Going to add also to other langauges

File: client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult_ja.properties (hunk 4 above)
Comment (andrea-patricelli) [reply client/idrepo/enduser/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult_ja.properties:17]: no, I think that this is an IDE issue. Going to replace with correct chars

File: client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/commons/RESTUtils.java (hunk 1 above)
Comment (andrea-patricelli) [reply client/idrepo/enduser/src/main/java/org/apache/syncope/client/enduser/commons/RESTUtils.java:28]: This class is mainly a wrapper for REST calls, going to refactor.

File: client/idrepo/enduser/pom.xml [hunk 6]
```diff
@@ -269,7 +269,6 @@ under the License.
             <configuration>
               <jvmArguments>
                 -Dwicket.core.settings.general.configuration-type=development
-                -XX:HotswapAgent=fatjar
```
Comment (ilgrosso) [inline client/idrepo/enduser/pom.xml:272]: Please re-add this option.

File: client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java [hunk 7]
```diff
@@ -197,8 +197,8 @@ protected void onClickTemplate(final AjaxRequestTarget target) {
         @Override
         protected void setWindowClosedReloadCallback(final BaseModal<?> modal) {
             modal.setWindowClosedCallback(target -> {
-                if (modal.getContent() instanceof ResultPage) {
-                    Serializable result = ResultPage.class.cast(modal.getContent()).getResult();
+                if (modal.getContent() instanceof ResultPanel) {
+                    Object result = ResultPanel.class.cast(modal.getContent()).getResult();
```
Comment (ilgrosso) [inline client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java:201]: Why not `Serializable` any more?

**Review by ilgrosso — CHANGES_REQUESTED** [review]

File: client/idrepo/enduser/pom.xml (hunk 6 above)
Comment (andrea-patricelli) [reply client/idrepo/enduser/pom.xml:272]: Sorry, missed to re-addd before push

Comment (ilgrosso) [reply client/idrepo/enduser/pom.xml:272]: ...or just install https://github.com/TravaOpenJDK/trava-jdk-11-dcevm and use that with `-Pdebug`

File: client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java (hunk 7 above)
Comment (andrea-patricelli) [reply client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java:201]: I use this panel like this `AbstractResultPanel<String, List<StatusBean>>` and List is not `Serializable`.

Comment (ilgrosso) [reply client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java:201]: Mmmh ok.
Please ensure that neither Core nor Console logs show serialization errors while using such panel.

Comment (andrea-patricelli) [reply client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java:201]: going to find a workaround

Comment (andrea-patricelli) [reply client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java:201]: Ok, let me check

File: client/idrepo/common-ui/src/main/java/org/apache/syncope/client/ui/commons/panels/SimpleListViewPanel.java [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+     * @param list      list of item.
+     * @param reference list item reference class.
+     * @param includes  Used to sort and restrict the set of bean's fields to be shown.
+     */
+    private SimpleListViewPanel(
+            final String id,
+            final List<T> list,
+            final Class<T> reference,
+            final List<String> includes,
+            final IModel<? extends Collection<T>> model) {
```
[the remark this answers is not in the document]
Comment (ilgrosso) [reply client/idrepo/common-ui/src/main/java/org/apache/syncope/client/ui/commons/panels/SimpleListViewPanel.java:66]: @andrea-patricelli please double-check if you really need the `model` parameter here

File: client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java (hunk 7 above)
Comment (andrea-patricelli) [reply client/idrepo/console/src/main/java/org/apache/syncope/client/console/pages/Realms.java:201]: Going to apply the same solution as in https://github.com/apache/syncope/blob/master/client/idm/console/src/main/java/org/apache/syncope/client/console/commons/IdMAnyDirectoryPanelAdditionalActionsProvider.java#L109

**Review by ilgrosso — APPROVED** [review]
LGTM @andrea-patricelli  please go ahead, squash and merge

**Pull request merged**